lib/preview_web/live/preview_live.ex

```diff
@@ -28,7 +36,7 @@ defmodule PreviewWeb.PreviewLive do
          canonical: canonical_url(params["package"], filename)
        )}
     else
-      {:ok, assign(socket, error: "TODO")}
+      raise Exception, plug_status: 404
```

Comment:
This does not seem like a great solution, maybe someone with better LiveView knowledge know a better way?

Reply:
Maybe [`redirect/2`](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#redirect/2) to a 404 page? But then again we won't have the 404 status. I think this is probably ok as is.

Reply:
This is correct. If mount fails during a LiveRedirect, we will refresh the page and go through the regular life-cycle where this will be handled by Plug.